src/geolocation.re

```diff
@@ -0,0 +1,104 @@
+type rnConfig;
+
+type currentPositionConfig;
+
+type watchPositionConfig;
+
+type watchId = int;
+
+type position = {
+  .
+  "coords": coords,
+  "timestamp": float,
+}
+and coords = {
+  .
+  "speed": int,
+  "longitude": float,
+  "latitude": float,
+  "accuracy": int,
+  "heading": int,
+};
+
+type error = {
+  .
+  "code": int,
+  "message": string,
+};
+
+[@bs.obj]
+external makeRnConfig : (~skipPermissionRequests: Js.boolean=?) => rnConfig =
+  "";
+
+[@bs.obj]
+external makeCurrentPositionConfig :
+  (~timeout: int=?, ~maximumAge: int=?, ~enableHighAccuracy: Js.boolean=?) =>
+  currentPositionConfig =
+  "";
+
+[@bs.obj]
+external makeWatchPositionConfig :
+  (
+    ~timeout: int=?,
+    ~maximumAge: int=?,
+    ~enableHighAccuracy: Js.boolean=?,
+    ~distanceFilter: int=?,
+    ~useSignificantChanges: Js.boolean=?
+  ) =>
+  watchPositionConfig =
+  "";
+
+[@bs.val "navigator.geolocation.setRNConfiguration"]
```

Comment:
I'd change this to 
```reason
[@bs.val] [@bs.scope ("navigator", "geolocation")] external setRNConfiguration : rnConfig => unit = "";
let setRNConfiguration(makeRnConfig(~skipPermissionRequests?));
```

Reply:
Same goes for all the nested bs.val bindings below

Reply:
I also feel like this shouldn't be a part of the GeoLocation module. Maybe it should be in some sort of root module or even just Configuration module or something like that.

Reply:
this is how the navigator global looks like (while using the simulator):

<img width="806" alt="react native debugger 2018-03-23 19-37-51" src="https://user-images.githubusercontent.com/1467712/37847640-d0323c68-2ed1-11e8-90d3-ab711020fdb4.png">

let's create `navigator.re` module, and add the `Geolocation` module there
we could add other bindings from the `navigator` global in next pull requests, what do you think about that? 😉

Reply:
Oh wait, I misread, didn't notice that setRNConfiguration was under the geolocation scope.

Sorry about that

But I still want to change it to be `[@bs.val] [@bs.scope ("navigator", "geolocation") ...` instead of having the dots there 😄